src/repl/repl.ts

```diff
@@ -46,12 +47,21 @@ function main() {
       if (err) {
         throw err
       }
-      startRepl(4, false, data)
+      startRepl(4, false,'interpreter', false, data)
     })
   } else {
     const chapter = process.argv.length > 2 ? parseInt(firstArg, 10) : 1
     const useSubst = process.argv.length > 3 ? process.argv[3] === 'subst' : false
-    startRepl(chapter, useSubst)
+    const ExecutionMethod = process.argv.length > 3 && process.argv[3] === 'interpreter' ? 'interpreter' : 'auto'
+
+    let useLazyEval = false;
+    if ((!useSubst && ExecutionMethod==='interpreter')&&
+        ((process.argv.length > 4 && process.argv[4]==='lazy')
+          ||(process.argv.length > 3 && process.argv[3]==='lazy')
+          ||(process.argv.length > 2 && process.argv[2]==='lazy'))){
+      useLazyEval = true;
+    }
```

Comment:
https://stackoverflow.com/questions/2430000/determine-if-string-is-in-list-in-javascript

Reply:
Thx! Actually I'm not very familiar with js

Reply:
just google lah :)))))